components/search_results/search_limits_banner.tsx

```diff
@@ -0,0 +1,127 @@
+background-color: rgba(var(--center-channel-text-rgb), 0.04);
+padding: 10px;
+margin: 10px;
+color: rgba(var(--center-channel-text-rgb), 0.72);
+font-weight: 400;
+font-size: 11px;
+line-height: 16px;
+letter-spacing: 0.02em;
+`;
+
+type Props = {
+    searchType: string;
+}
+
+const FILES_SEARCH_TYPE = 'files';
+const MESSAGES_SEARCH_TYPE = 'messages';
+
+function SearchLimitsBanner(props: Props) {
+    const {formatMessage, formatNumber} = useIntl();
+    const openPricingModal = useOpenPricingModal();
+    const usage = useGetUsage();
+    const isAdminUser = isAdmin(useSelector(getCurrentUser).roles);
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const product = useSelector(selectSubscriptionProduct);
+    const isStarter = product?.sku === CloudProducts.STARTER;
+
+    if (!isCloud || !isStarter) {
+        return null;
+    }
+
+    const currentFileStorageUsage = usage?.files?.totalStorage || 0;
+    const currentMessagesUsage = usage?.messages?.history || 0;
+
+    let ctaAction = formatMessage({
+        id: 'workspace_limits.search_limit.view_plans',
+        defaultMessage: 'View plans',
+    });
+
+    if (isAdminUser) {
+        ctaAction = formatMessage({
+            id: 'workspace_limits.search_limit.upgrade_now',
+            defaultMessage: 'Upgrade now',
+        });
+    }
+
+    const renderBanner = (bannerText: React.ReactNode, id: string) => {
+        return (<StyledDiv id={id}>
+            <InnerDiv>
+                <i className='icon-eye-off-outline'/>
+                <span>{bannerText}</span>
+            </InnerDiv>
+        </StyledDiv>);
+    };
+
+    switch (props.searchType) {
+    case FILES_SEARCH_TYPE:
+        if (!(currentFileStorageUsage > fallbackStarterLimits.files.totalStorage)) {
+            return null;
+        }
+        return renderBanner(formatMessage({
+            id: 'workspace_limits.search_files_limit.banner_text',
+            defaultMessage: 'Some older files may not be shown because your workspace has met its file storage limit of {storage}. <a>{ctaAction}</a>',
+        }, {
+            ctaAction,
+            storage: asGBString(usage.files.totalStorage, formatNumber),
+            a: (chunks: React.ReactNode | React.ReactNodeArray) => (
+                <StyledA
+                    onClick={openPricingModal}
+                >
+                    {chunks}
+                </StyledA>
+            ),
+        }), `${FILES_SEARCH_TYPE}_search_limits_banner`);
+
+    case MESSAGES_SEARCH_TYPE:
+        if (!(currentMessagesUsage > fallbackStarterLimits.messages.history)) {
+            return null;
+        }
+        return renderBanner(formatMessage({
+            id: 'workspace_limits.search_message_limit.banner_text',
+            defaultMessage: 'Some older messages may not be shown because your workspace has over {messages} messages. <a>{ctaAction}</a>',
+        }, {
+            ctaAction,
+            messages: formatNumber(usage.messages.history),
```

Comment:
The value of `messages` passed in should be the subscription based limit, not the usage.